WordPress/gutenberg
# Improve file block accessibility by adding aria-describedby to download button

## Description

This PR adds an id to the file link and adds that id as an `aria-describedby` on the download button. This announces the Download button in screen readers as "Download, [content from the file link text]".

![fileblock](https://user-images.githubusercontent.com/1689238/111543866-07048b00-874a-11eb-9c59-176c087563a3.gif)

For example, in this gif with "captions-1" as the link text, it announces the Download button as "Download, captions-1."

To avoid duplicate ids (and the potential for the wrong file name being linked to the download button), I used the `clientId` [method suggested here](https://github.com/WordPress/gutenberg/issues/17246#issuecomment-616111416).

Fixes #22908
Updates #22914

## How has this been tested?




1. Checkout this PR: `git fetch origin pull/28062/head:pr/28062 && git checkout pr/28062`
2. Add a new file block to a post:
<img width="387" alt="Screen Shot 2021-03-17 at 5 59 01 PM" src="https://user-images.githubusercontent.com/1689238/111544203-7e3a1f00-874a-11eb-88fe-e4d967bd79b5.png">

3. Upload a file and publish or preview the post.
4. Navigate the post with a screen reader. The Download button should be announced as "Download," then the file name.

## Types of changes




This change fixes an accessibility issue, and requires a deprecation entry and new fixtures.

## Checklist:
